File: Parallel-Csharp/src/Parallel/Vector/Vector3.cs
namespace Parallel.Vector;

public struct Vector3
{
    private double _x;

    private double _y;

    private double _z;

    public double X
    {
        get => _x;
        set => _x = value;
    }

    public double Y
    {
        get => _y;
        set => _y = value;
    }

    public Vector3(double x, double y, double z)
    {
        _x = x;
        _y = y;
        _z = z;
    }

    public double Distance(Vector3 other)
    {
        double distX = other._x - _x;
        double distY = other._y - _y;
        double distZ = other._z - _z;
        return Math.Sqrt(distX * distX + distY * distY + distZ * distZ);
    }
}

File: Parallel-Java/src/parallel/vector/Vector3.java
package parallel.vector;

public class Vector3 {
    private double x;

    private double y;

    private double z;

    public Vector3(double x, double y, double z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }

    public double getX() {
        return x;
    }

    public void setX(double x) {
        this.x = x;
    }

    public double getY() {
        return y;
    }

    public void setY(double y) {
        this.y = y;
    }

    public double getZ() {
        return z;
    }

    public void setZ(double z) {
        this.z = z;
    }

    public double distance(Vector3 other)
    {
        double distX = other.x - x;
        double distY = other.y - y;
        double distZ = other.z - z;
        return Math.hypot(Math.hypot(distX, distY), distZ);
    }
}
